File: db4oj/core/src/com/db4o/cs/messages/MTaIsDeleted.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.cs.messages;

import com.db4o.cs.*;

public final class MTaIsDeleted extends MsgD {
	public final boolean processAtServer(YapServerThread serverThread) {
		synchronized (streamLock()) {
			boolean isDeleted = getTransaction().isDeleted(this.readInt());
			int ret = isDeleted ? 1 : 0;
			serverThread.write(Msg.TA_IS_DELETED.getWriterForInt(getTransaction(), ret));
		}
		return true;
	}
}

File: db4on/core/src/core/com/db4o/cs/messages/MTaIsDeleted.cs
namespace com.db4o.cs.messages
{
	public sealed class MTaIsDeleted : com.db4o.cs.messages.MsgD
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			com.db4o.YapStream stream = GetStream();
			lock (stream.i_lock)
			{
				bool isDeleted = GetTransaction().IsDeleted(this.ReadInt());
				int ret = isDeleted ? 1 : 0;
				com.db4o.cs.messages.Msg.TA_IS_DELETED.GetWriterForInt(GetTransaction(), ret).Write
					(stream, sock);
			}
			return true;
		}
	}
}
